### PR TITLE
Zarr conversion + data loader

### PR DESCRIPTION
**Pull request by forklady42**
https://github.com/Quantum-Accelerators/electrai/issues/14

First take on a script for converting MP charge density data to zarr and a corresponding zarr dataloader. Still doing some clean up, but creating a draft PR to ask clarifying questions.

For this first take, sticking to the existing dataloader interface, so that this can be tried as a drop in replacement. This does not yet take advantage of lazy loading. That should follow in a future PR.

# Example chgcar comparison
When converted to zarr, the 10 example chgcar *.json.gz files are a total of 92 MB. Converted to zarr, they are 51 MB, a ~45% reduction. This means that even besides gains with the dataloader, we will see improvement on disk due to zarr's compression.

I tested the training script locally with the 10 converted examples to confirm the zarr dataloader worked. It ran twice as fast as the training script using the RhoData dataloader. The exact speed up will look different with larger amounts of data, as functions other than data loading come into play. For now, focus on the directionality. It also used less memory.

### CODE REVIEW

**Comment by forklady42** [comment]
@hanaol this is now ready for review. I've changed the head of this branch to https://github.com/Quantum-Accelerators/electrai/pull/27 as one of the dependencies requires Python version >3.9.

File: src/electrai/configs/config_zarr.yaml
Comment (hanaol) [inline src/electrai/configs/config_zarr.yaml]: Can you modify this configuration according to the refactored code and #26?

File: src/electrai/configs/config_zarr.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Dataset / loader parameters for Zarr format
+dataset_name: "zarr_s3_data"
```
Comment (hanaol) [inline src/electrai/configs/config_zarr.yaml]: We may add additional datasets to the registry later, so it would be helpful to include mp in the dataset name—for example, mp_zarr_s3_data.

**Comment by forklady42** [comment]
@hanaol I updated the zarr dataloader terms to specify `mp`.

**Comment by hanaol** [comment]
lgtm!
